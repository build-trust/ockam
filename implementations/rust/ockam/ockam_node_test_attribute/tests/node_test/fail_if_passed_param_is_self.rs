// This test checks that #[ockam_node_test_attribute::node_test] causes a compile time error
// if the function is passed a `self` param (thus making it a
// `Receiver` function.

#[ockam_node_test_attribute::node_test]
async fn my_test(self) -> ockam::Result<()> {}
