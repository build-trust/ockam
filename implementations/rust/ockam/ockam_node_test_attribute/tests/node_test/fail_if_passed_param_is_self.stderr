error: Input argument should be of type `ockam::Context`
 --> tests/node_test/fail_if_passed_param_is_self.rs:6:18
  |
6 | async fn my_test(self) -> ockam::Result<()> {}
  |                  ^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/node_test/fail_if_passed_param_is_self.rs:5:1
  |
5 | / #[ockam_node_test_attribute::node_test]
6 | | async fn my_test(self) -> ockam::Result<()> {}
  | |______________________________________________^ consider adding a `main` function to `$DIR/tests/node_test/fail_if_passed_param_is_self.rs`
